### PR TITLE
docs: move docs

### PR DESCRIPTION
**Pull request by yuyinws**
### 🔗 Linked issue

https://github.com/unplugin/docs/issues/7

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR includes:
1. Move [unplugin/docs](https://github.com/unplugin/docs/) to this repo.
2. Add a new GitHub workflows file. Ensure that Vercel automatic build is triggered every time the main branch is updated.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by yuyinws** [comment]
@sxzz Hi, Once this PR merge. I will dm you in discord about the GitHub action secrets like `VERCEL_TOKEN`.

**Comment by sxzz** [comment]
@antfu Could you please grant me the admin access of repo.

File: .github/workflows/vercel-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Vercel Deploy
+
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+
+on:
+  schedule:
```
Comment (sxzz) [inline .github/workflows/vercel-deploy.yml:8]: Is that necessary to re-deploy everyday?
I think `push` and `workflow_dispatch` are enough.

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ignore-workspace-root-check=true
+shamefully-hoist=true
```
Comment (sxzz) [inline .npmrc:2]: Can we avoid `shamefully-hoist`?

File: tsconfig.json [hunk 3]
```diff
@@ -21,6 +21,9 @@
   },
   "include": [
     "src",
-    "test"
+    "test",
+    "docs/.vitepress/*.ts",
```
Comment (sxzz) [inline tsconfig.json:25]:
```suggestion
    "docs/.vitepress",
```

This single line should be enough, right?

File: .npmrc (hunk 2 above)
Comment (yuyinws) [reply .npmrc:2]: After remove it, vitepress build get [some error](https://github.com/WestonSeyler/unplugin/actions/runs/8197271878/job/22418976460#step:6:109) throw by `twoslash`.

Comment (sxzz) [reply .npmrc:2]: Because `unplugin` itself wasn't built.

**Comment by sxzz** [comment]
I plan to move it to Netlify by using Anthony's or Unjs org's account.

**Comment by yuyinws** [comment]
After the new site is available, I will let the old domain redirect to the new one.

**Review by antfu — APPROVED** [review]

**Pull request merged**